reference/tugboat-configuration/index.md

```diff
@@ -0,0 +1,194 @@
+The name must be unique in the `services` list, and must also be a valid host
+name containing only the characters a-z, 0-9, and hyphen (`[a-z0-9-]`).
+
+The following attributes control how a service is built:
+
+| Key                            | Type    | Description                                         |
+| :----------------------------- | :------ | :-------------------------------------------------- |
+| [image](#image)                | String  | The Docker image to use for this service            |
+| [checkout](#checkout)          | Boolean | Whether to clone the git repository to this service |
+| [checkout_path](#checkoutpath) | String  | Where to clone the git repository                   |
+| [depends](#depends)            | List    | List of other services that this service depends on |
+| [commands](#commands)          | List    | List of commands to run for various build stages    |
+
+The following attributes configure how the Tugboat proxy routes HTTP requests to
+the service:
+
+| Key                              | Type    | Description                                               |
+| :------------------------------- | :------ | :-------------------------------------------------------- |
+| [expose](#expose)                | Integer | Which port the service should expose to the Tugboat proxy |
+| [default](#default)              | Boolean | Whether this is the default service for a preview.        |
+| [http](#http)                    | Boolean | Whether the service should be available via HTTP          |
+| [https](#https)                  | Boolean | Whether the service should be available via HTTPS         |
+| [https_redirect](#httpsredirect) | Boolean | Whether HTTP requests should be redirected to HTTPS       |
+| [domain](#domain)                | String  | A custom domain for Tugboat to generate URLs with         |
+| [subpath](#subpath)              | Boolean | Whether subpath URLs should be generated for this service |
+| [subpath_map](#subpathmap)       | Boolean | Whether to map the generated subpath to "/"               |
+
+---
+
+## `checkout`
+
+* **Type:** Boolean
+* **Default:** `false` (`true` when `default: true` is set)
+* **Required:** No
+
+Whether or not this service should have a copy of the git repository cloned into
+it.
+
+## `checkout_path`
+
+* **Type:** String
+* **Default:** `/var/lib/tugboat`
+* **Required:** No
+
+Specifies the path where the git repository should be cloned if `checkout: true`
+is set. If this path already exists, the clone will fail.
+
+## `commands`
+
+* **Type:** List
+* **Default:** _no default_
+* **Required:** No
+
+A set of commands to run during various points in a preview's lifecycle. These
+commands are divided into the following stages. Each stage consists of a list of
+commands. Each command is run in its own context, so things like changing
+directories does not "stick" between commands. If that behavior is requried, an
+external script should be called.
```

Comment:
Where do I find how to link to an external script?

Reply:
https://pr32-ltcrmewfcubcfohtf6ajgyszeqq4cadb.tugboat.qa/configuring-tugboat/#commands